### ISSUE TITLE
version-control-diff right side on Emacs GUI and not working properly on Emacsclient GUI?

### ISSUE THREAD

**Issue by myme5261314** [syl20bnr/spacemacs#5821]
I'm not sure why my version-control-diff symbols appear on the right side of the Emacs GUI like below:

`
#### System Info
- OS: gnu/linux
- Emacs: 24.5.1
- Spacemacs: 0.105.18
- Spacemacs branch: develop (rev. b123b21)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(html spacemacs-helm spacemacs-ivy better-defaults github
      (version-control :variables version-control-diff-tool 'git-gutter version-control-global-margin t)
      semantic markdown
      (vinegar :variables vinegar-reuse-dired-buffer t)
      org prodigy search-engine
      (syntax-checking :variables syntax-checking-enable-by-default nil)
      (spell-checking :variables spell-checking-enable-by-default nil)
      yaml python lua html command-log emacs-lisp emoji fasd deft ranger gtags
      (spacemacs-layouts :variables layouts-enable-autosave t layouts-autosave-delay 300)
      (colors :variables colors-enable-nyan-cat-progress-bar t)
      (git :variables git-magit-status-fullscreen t magit-push-always-verify nil magit-save-repository-buffers 'dontask magit-revert-buffers 'silent magit-refs-show-commit-count 'all magit-revision-show-gravatars nil)
      (ibuffer :variables ibuffer-group-buffers-by 'projects)
      (c-c++ :variables c-c++-default-mode-for-headers 'c++-mode)
      (auto-completion :variables auto-completion-enable-sort-by-usage t auto-completion-enable-snippets-in-popup t)
      (shell :variables shell-default-position 'full shell-default-shell 'ansi-term shell-default-term-shell "/bin/zsh")
      (chinese :variables chinese-default-input-method 'pinyin chinese-enable-fcitx t chinese-enable-youdao-dict t)
      emacs-lisp markdown org dash my)
```
![image](https://cloud.githubusercontent.com/assets/1814831/14592988/ac5a2f6c-0557-11e6-9c6c-3e79f8ef69d8.png)

#### System Info
- OS: gnu/linux
- Emacs: 24.5.1
- Spacemacs: 0.105.18
- Spacemacs branch: develop (rev. b123b21)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(html spacemacs-helm spacemacs-ivy better-defaults github
      (version-control :variables version-control-diff-tool 'git-gutter+ version-control-global-margin t)
      semantic markdown
      (vinegar :variables vinegar-reuse-dired-buffer t)
      org prodigy search-engine
      (syntax-checking :variables syntax-checking-enable-by-default nil)
      (spell-checking :variables spell-checking-enable-by-default nil)
      yaml python lua html command-log emacs-lisp emoji fasd deft ranger gtags
      (spacemacs-layouts :variables layouts-enable-autosave t layouts-autosave-delay 300)
      (colors :variables colors-enable-nyan-cat-progress-bar t)
      (git :variables git-magit-status-fullscreen t magit-push-always-verify nil magit-save-repository-buffers 'dontask magit-revert-buffers 'silent magit-refs-show-commit-count 'all magit-revision-show-gravatars nil)
      (ibuffer :variables ibuffer-group-buffers-by 'projects)
      (c-c++ :variables c-c++-default-mode-for-headers 'c++-mode)
      (auto-completion :variables auto-completion-enable-sort-by-usage t auto-completion-enable-snippets-in-popup t)
      (shell :variables shell-default-position 'full shell-default-shell 'ansi-term shell-default-term-shell "/bin/zsh")
      (chinese :variables chinese-default-input-method 'pinyin chinese-enable-fcitx t chinese-enable-youdao-dict t)
      emacs-lisp markdown org dash my)
```
![image](https://cloud.githubusercontent.com/assets/1814831/14593019/13441530-0558-11e6-8162-524fa68bbc27.png)

----
And with Emacsclient, the git-gutter+ seems not working properly.
#### System Info
- OS: gnu/linux
- Emacs: 24.5.1
- Spacemacs: 0.105.18
- Spacemacs branch: develop (rev. b123b21)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(html spacemacs-helm spacemacs-ivy better-defaults github
      (version-control :variables version-control-diff-tool 'git-gutter+ version-control-global-margin t)
      semantic markdown
      (vinegar :variables vinegar-reuse-dired-buffer t)
      org prodigy search-engine
      (syntax-checking :variables syntax-checking-enable-by-default nil)
      (spell-checking :variables spell-checking-enable-by-default nil)
      yaml python lua html command-log emacs-lisp emoji fasd deft ranger gtags
      (spacemacs-layouts :variables layouts-enable-autosave t layouts-autosave-delay 300)
      (colors :variables colors-enable-nyan-cat-progress-bar t)
      (git :variables git-magit-status-fullscreen t magit-push-always-verify nil magit-save-repository-buffers 'dontask magit-revert-buffers 'silent magit-refs-show-commit-count 'all magit-revision-show-gravatars nil)
      (ibuffer :variables ibuffer-group-buffers-by 'projects)
      (c-c++ :variables c-c++-default-mode-for-headers 'c++-mode)
      (auto-completion :variables auto-completion-enable-sort-by-usage t auto-completion-enable-snippets-in-popup t)
      (shell :variables shell-default-position 'full shell-default-shell 'ansi-term shell-default-term-shell "/bin/zsh")
      (chinese :variables chinese-default-input-method 'pinyin chinese-enable-fcitx t chinese-enable-youdao-dict t)
      emacs-lisp markdown org dash my)
```
![image](https://cloud.githubusercontent.com/assets/1814831/14593100/d03fe452-0558-11e6-93a0-76f8f7ceea8a.png)

------
But, git-gutter seems to work properly under Emacsclient.
#### System Info
- OS: gnu/linux
- Emacs: 24.5.1
- Spacemacs: 0.105.18
- Spacemacs branch: develop (rev. b123b21)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(html spacemacs-helm spacemacs-ivy better-defaults github
      (version-control :variables version-control-diff-tool 'git-gutter version-control-global-margin t)
      semantic markdown
      (vinegar :variables vinegar-reuse-dired-buffer t)
      org prodigy search-engine
      (syntax-checking :variables syntax-checking-enable-by-default nil)
      (spell-checking :variables spell-checking-enable-by-default nil)
      yaml python lua html command-log emacs-lisp emoji fasd deft ranger gtags
      (spacemacs-layouts :variables layouts-enable-autosave t layouts-autosave-delay 300)
      (colors :variables colors-enable-nyan-cat-progress-bar t)
      (git :variables git-magit-status-fullscreen t magit-push-always-verify nil magit-save-repository-buffers 'dontask magit-revert-buffers 'silent magit-refs-show-commit-count 'all magit-revision-show-gravatars nil)
      (ibuffer :variables ibuffer-group-buffers-by 'projects)
      (c-c++ :variables c-c++-default-mode-for-headers 'c++-mode)
      (auto-completion :variables auto-completion-enable-sort-by-usage t auto-completion-enable-snippets-in-popup t)
      (shell :variables shell-default-position 'full shell-default-shell 'ansi-term shell-default-term-shell "/bin/zsh")
      (chinese :variables chinese-default-input-method 'pinyin chinese-enable-fcitx t chinese-enable-youdao-dict t)
      emacs-lisp markdown org dash my)
```

![image](https://cloud.githubusercontent.com/assets/1814831/14593157/7a863790-0559-11e6-8f35-6cba35467c01.png)

**Comment by bb4242** [comment]
I've noticed this as well - there is a slightly different visual appearance between emacs and emacsclient.  For me, the diff symbols appear on the right in emacs, and on the left in emacsclient.  Also, in emacsclient, "line modified" symbols don't appear at all, only "line added" and "line deleted".

Spaceline also has a slightly different appearance between emacs and emacsclient.  See attached screenshots.

emacs:
![2016-05-13-102710_1920x1080_scrot](https://cloud.githubusercontent.com/assets/10068296/15256307/d99085a6-18f5-11e6-8b95-2068c7a5e633.png)

emacsclient:
![2016-05-13-102748_1920x1080_scrot](https://cloud.githubusercontent.com/assets/10068296/15256312/dd80c342-18f5-11e6-9627-f57c0ee190a3.png)

**Comment by fardog** [comment]
I'm also experiencing this issue; git-gutter+ looks correct when emacs is run normally:

![2016-07-20-061429_1596x857_scrot](https://cloud.githubusercontent.com/assets/2207657/16987825/f809df6e-4e42-11e6-9914-10930ee86040.png)

However when I run a `emacs --daemon` and connect with `emacsclient -nc`, the gutters appear on the wrong side, and as blocks which disappear as I scroll:

![2016-07-20-061146_1596x857_scrot](https://cloud.githubusercontent.com/assets/2207657/16987914/51550bca-4e43-11e6-869f-1509b511fe86.png)

- OS: gnu/linux
- Emacs: 24.5.1
- Spacemacs: 0.105.21
- Spacemacs branch: master (rev. 0283f64)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(emacs-lisp clojure python javascript markdown go auto-completion syntax-checking version-control git github
            (shell :variables shell-default-shell 'ansi-term))
```

Let me know if there's other useful information I could provide!

**Comment by TheBB** [comment]
@fardog Can you try in develop, possibly with https://github.com/syl20bnr/spacemacs/pull/6642?

**Comment by fardog** [comment]
@TheBB unfortunately, neither the `develop` branch nor #6642 fixed the issue. I have been able to fix the issue by adding the following to my `dotspacemacs-configuration-layers` list, however:

```elisp
     (version-control :variables version-control-diff-tool 'git-gutter version-control-global-margin t)
```

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.
